ui/ui-components/contexts/grouparooModelContext.tsx

```diff
@@ -0,0 +1,15 @@
+import { createContext, useContext } from "react";
+import { Models } from "../utils/apiData";
+
+export const GrouparooModelContext =
+  createContext<Models.GrouparooModelType>(undefined);
+
+export const useGrouparooModelContext = () => useContext(GrouparooModelContext);
+
+export const GrouparooModelContextProvider: React.FC<{
+  model: Models.GrouparooModelType;
+}> = ({ model, children }) => (
+  <GrouparooModelContext.Provider value={model}>
+    {children}
+  </GrouparooModelContext.Provider>
+);
```

Comment:
This is another new pattern I'm introducing. This sets up everything that is needed for a context such that the context user just needs to either use the rich `GrouparooModelContextProvider` provider or the rich `useGrouparooModelContext` hook. This is pretty simple but as more logic may be added the provider can be extended to manage the value of the provider.